lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -333,6 +334,54 @@ UsdTimeCode getTime(const Ufe::Path& path)
     return proxyShape->getTime();
 }
 
+TfTokenVector getProxyShapePurposes(const Ufe::Path& path)
```

Comment:
Copy-pasted and adapted from getTime().  Arguably, getTime() itself has duplication from the stage map, and all calls to nameToDagPath() in the existing code base should be examined for performance and centralized.  This is out of scope for the current branch.